From: sunshineinabox <aqemail@gmail.com>
Date: Mon, 27 Jul 2026 20:44:11 -0700
Subject: [PATCH] ASoC: qcom: sc8280xp: apply codec_dai_fmt to all codec DAIs

With MI2S interfaces, multiple codecs typically share one bus.
All codecs on the bus should receive the same format configuration,
so apply codec_dai_fmt to all of them instead of only the first.

Adapted from Val Packett's equivalent sm8250.c change.
---
 sound/soc/qcom/sc8280xp.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/sound/soc/qcom/sc8280xp.c b/sound/soc/qcom/sc8280xp.c
index db46d78..2911448 100644
--- a/sound/soc/qcom/sc8280xp.c
+++ b/sound/soc/qcom/sc8280xp.c
@@ -76,13 +76,16 @@ static int sc8280xp_snd_startup(struct snd_pcm_substream *substream)
 	struct snd_soc_pcm_runtime *rtd = snd_soc_substream_to_rtd(substream);
 	struct sc8280xp_snd_data *pdata = snd_soc_card_get_drvdata(rtd->card);
 	struct snd_soc_dai *cpu_dai = snd_soc_rtd_to_cpu(rtd, 0);
-	struct snd_soc_dai *codec_dai = snd_soc_rtd_to_codec(rtd, 0);
+	struct snd_soc_dai *codec_dai;
+	int i;
 
 	switch (cpu_dai->id) {
 	case PRIMARY_MI2S_RX:
 		clk_prepare_enable(pdata->i2s_clk);
 		snd_soc_dai_set_fmt(cpu_dai, fmt);
-		snd_soc_dai_set_fmt(codec_dai, codec_dai_fmt);
+		/* All codecs sharing the bus need the same format */
+		for_each_rtd_codec_dais(rtd, i, codec_dai)
+			snd_soc_dai_set_fmt(codec_dai, codec_dai_fmt);
 		break;
 	default:
 		break;
-- 
2.55.0

